transport/http/config.go

```diff
@@ -0,0 +1,110 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package http
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/x/config"
+)
+
+// TransportConfig TODO
+type TransportConfig struct {
+	KeepAlive time.Duration `config:"keepAlive"`
+}
+
+// BuildTransport TODO
+func (tc *TransportConfig) BuildTransport() (transport.Transport, error) {
+	var opts []TransportOption
+	if tc.KeepAlive > 0 {
+		opts = append(opts, KeepAlive(tc.KeepAlive))
+	}
+
+	return NewTransport(opts...), nil
+}
+
+// InboundConfig TODO
+type InboundConfig struct {
+	Address string `config:"address"`
+}
+
+// BuildInbound TODO
+func (ic *InboundConfig) BuildInbound(t transport.Transport) (transport.Inbound, error) {
+	if ic.Address == "" {
+		return nil, fmt.Errorf("inbound address is required")
+	}
+
+	return t.(*Transport).NewInbound(ic.Address), nil
+}
+
+// OutboundConfig TODO
+type OutboundConfig struct {
+	Host   string `config:"host"`
+	Port   int    `config:"port"`
+	Path   string `config:"path"`
+	Scheme string `config:"scheme"`
+}
+
+// BuildUnaryOutbound TODO
+func (oc *OutboundConfig) BuildUnaryOutbound(t transport.Transport) (transport.UnaryOutbound, error) {
+	return oc.buildOutbound(t)
+}
+
+// BuildOnewayOutbound TODO
+func (oc *OutboundConfig) BuildOnewayOutbound(t transport.Transport) (transport.OnewayOutbound, error) {
+	return oc.buildOutbound(t)
+}
+
+func (oc *OutboundConfig) buildOutbound(t transport.Transport) (*Outbound, error) {
+	if oc.Host == "" {
+		return nil, errors.New("outbound host is required")
+	}
+	if oc.Port == 0 {
+		return nil, errors.New("outbound port is required")
+	}
+
+	if oc.Scheme == "" {
+		oc.Scheme = "http"
+	}
+
+	if oc.Path != "" && oc.Path[0] == '/' {
+		oc.Path = oc.Path[1:]
+	}
+
+	url := fmt.Sprintf("%s://%s:%d/%s", oc.Scheme, oc.Host, oc.Port, oc.Path)
+	return t.(*Transport).NewSingleOutbound(url), nil
+}
+
+// TODO: presets
+
+// RegisterTransport TODO
+func RegisterTransport(l *config.Loader) error {
+	return l.RegisterTransport(config.TransportSpec{
```

Comment:
In order to allow us to have continual customizability, we might want to have a "http.TransportSpec" function we can grab.

Additionally, in order to allow "pre-filling" of configuration for an outbound/inbound/etc we might want to include a "with" option for outbounds/inbounds/etc, that way we can inject custom config logic for internal use cases.

So for registry we could do something like this:
```
loader :=  getLoader()
httpSpec := http.TransportSpec(
   http.CustomOutbound("dns", MyDNSOutboundBuilder)
   http.CustomOutbound("file", MyFileOutboundBuilder)
   ...
)
loader.RegisterTransport(httpSpec)
```
and in our config we can do:
```
outbounds:
  myoutbound:
    http:
      with: 
        dns: www.myurl.com # uses the MyDNSOutboundBuilder to build the outbound.
```
This will require the outbound config to keep track of what outbounds have been registered to use the "with" command, but it gives us a lot of flexibility with our custom routing use cases.